### PR TITLE
[Popup] Ground work for common popup

### PR DESCRIPTION
**Pull request by avanderhoorn**
Basic ground work for common hud pop.

### CODE REVIEW

File: src/proxy/timing.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const timingsRaw = (window.performance ||
     window.msPerformanceperformance ||
     window.webkitPerformanceperformance ||
     {}).timing;
-let timingIncomplete = false;
+let timings = undefined;
```
Comment (davidkpiano) [inline src/proxy/timing.js]: Is this just for memoization? (commenting here for reference for future refactor)

File: src/views/ajax.js [hunk 2]
```diff
@@ -7,43 +7,89 @@ const state = {
...
     preRenderCache: [],
-    currentTimeout: undefined,
-    summaryStack: []
+    summary: {
+        countId: 'glimpse-ajax-count',
+        rowsId: 'glimpse-ajax-rows',
+        timeout: undefined,
+        stack: [],
+        template: rowTemplate,
+        length: -1
```
Comment (davidkpiano) [inline src/views/ajax.js:16]: What does this refer to?

Comment (davidkpiano) [reply src/views/ajax.js:16]: (and `popup.length` below... confused about the negative values)

Comment (avanderhoorn) [reply src/views/ajax.js:16]: Its used for how many records should be preserved in the stack.

File: src/proxy/timing.js (hunk 1 above)
Comment (avanderhoorn) [reply src/proxy/timing.js]: yep.

File: src/views/ajax.js [hunk 3]
```diff
@@ -7,43 +7,89 @@ const state = {
...
-    //manage counter value
-    var counter = document.getElementById('glimpse-ajax-count');
+    updateCounter(state.summary);
+    updateCounter(state.popup);
+
+    updateView(state.summary, request);
+    updateView(state.popup, request);
+}
+
+function updateCounter(details) {
```
Comment (davidkpiano) [inline src/views/ajax.js]: Let's make this more pure:

```js
function updateCounter({ details, count }) {
  // ...
}
```

and pass the `state.count` in from `function update(...) { ... }`

Comment (avanderhoorn) [reply src/views/ajax.js]: Sounds good.

**Comment by davidkpiano** [comment]
LGTM overall, some nits that move towards these goals:

- making the functions as pure as possible
- separating state changes from UI changes
- isolating UI side-effects to one "sink"

**Review by davidkpiano — APPROVED** [review]

**Comment by davidkpiano** [comment]
LGTM!

**Pull request merged**